File: 150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    long long evalRPN(vector<string>& tokens) {
        string s = tokens.back();
        tokens.pop_back();
        if(s != "+" && s != "-" && s != "*" && s != "/") return stoi(s);
        else {
            long long a = evalRPN(tokens);
            long long b = evalRPN(tokens);

            if(s == "+") return a + b;
            else if (s == "-") return b - a;
            else if (s == "/") return b / a;
            else return a * b;
        }
    }
};

File: 150-evaluate-reverse-polish-notation.py

class Solution(object):
    def evalRPN(self, tokens):
        """
        :type tokens: List[str]
        :rtype: int
        """
        s = tokens[len(tokens) - 1]
        tokens.pop()
        if(s not in '+-/*'): return int(s)
        else:
            a = self.evalRPN(tokens)
            b = self.evalRPN(tokens)
            if(s == "+"): return a + b
            elif(s == "*"): return a * b
            elif(s == "-"): return b - a
            else: return int(float(b)/a)
        
